client/me/account/main.jsx

```diff
@@ -84,25 +82,58 @@ const Account = React.createClass( {
 		debug( this.constructor.displayName + ' component is unmounting.' );
 	},
 
-	updateLanguage() {
-		let valueLink = this.valueLink( 'language' );
+	getUserSetting( settingName ) {
+		return this.props.userSettings.getSetting( settingName );
```

Comment:
> There is a lot of method calls on the this.props.userSettings and this.props.username in this module, so it would make sense to wrap them all in invoke.
>
> The only place where this component is instantiated is the controller.js in the same directory. > These two components are tightly-coupled "friends", so I think that the client/me/account/main.jsx component shouldn't need to worry if its props have the right type. > What do you think?

we can be easier with props, but we should also be declaring `propTypes` and listing which ones are required. `this.props` is safe and so is `this.props.someValue` because even if `someValue` doesn't exist we are fine. what _isn't_ okay is `this.props.someValue.someProperty` because if `someValue` isn't an object now we have an actual browser error